pkg/eval/builtin_fn_misc.go

```diff
@@ -160,7 +161,30 @@ import (
 // bar
 // ```
 
-// TODO(xiaq): Document esleep.
+//elvdoc:fn sleep
+//
+// ```elvish
+// sleep $duration
+// ```
+//
+// Sleep for the specified duration.  A duration string is a possibly signed
+// sequence of decimal numbers, each with optional fraction and a unit suffix,
+// such as "300ms", "-1.5h" or "2h45m". Valid time units are "ns", "us" (or
```

Comment:
Since you mentioned a negative duration as an example, also document the behavior when given a negative duration.